filament/src/FrameHistory.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TNT_FILAMENT_FRAMEHISTORY_H
+#define TNT_FILAMENT_FRAMEHISTORY_H
+
+namespace filament {
+
+// This is where we store all the history of a frame
+struct FrameHistoryEntry {
+    // history data goes here
+};
+
+/*
+ * This is a very simple FIFO that stores the history of previous frames.
+ */
+template<typename T, size_t SIZE = 1u>
+class TFrameHistory {
```

Comment:
Currently, this is a very generic FIFO.  Since it is an internal class that is specifically called "FrameHistory", I would consider publicizing `commit()`, adding a public `drain()` method, removing the friend declaration, and removing the two new pass-through methods in FView in favor of `FView::getFrameHistory()`.

I would also consider moving the default size of "1" from the template arg to the "using" alias.

Reply:
all sounds good. will do.

Reply:
btw, we can't remove the "pass-through" because in the future they won't be pass-through. This is where the clean-up of the resources stored in FrameEntry will happen. The dtor can't do it alone because it needs the allocator that might have been used for allocating the resources in the first place.

Reply:
Ah ok, understood.